PhysicsTools/NanoAOD/plugins/BJetEnergyRegressionMVA.cc

```diff
@@ -42,65 +43,95 @@ class BJetEnergyRegressionMVA : public BaseMVAValueMapProducer<pat::Jet> {
 	  void readAdditionalCollections(edm::Event&iEvent, const edm::EventSetup&) override {
 		iEvent.getByToken(pvsrc_, pvs_);
 		iEvent.getByToken(svsrc_, svs_);
-	  }
-
-          void fillAdditionalVariables(const pat::Jet&j)  override {
-		this->setValue("nPVs",pvs_->size());
-		BaseMVAValueMapProducer<pat::Jet>::setValue("Jet_leptonPtRel",0);
-
-		if(!j.overlaps("muons").empty()) { 
-			const auto *lep=dynamic_cast<const pat::Muon *>(&*j.overlaps("muons")[0]);
-			if(lep!=nullptr) {BaseMVAValueMapProducer<pat::Jet>::setValue("Jet_leptonPtRel",lep->userFloat("ptRel"));}
-		}
-		else if(!j.overlaps("electrons").empty()) {
-			const auto *lep=dynamic_cast<const pat::Electron *>(&*j.overlaps("electrons")[0]);
-			if(lep!=nullptr) {BaseMVAValueMapProducer<pat::Jet>::setValue("Jet_leptonPtRel",lep->userFloat("ptRel"));}
-		}
-		
-		float ptMax=0;
-		for(const auto & d : j.daughterPtrVector()){if(d->pt()>ptMax) ptMax=d->pt();}
-		BaseMVAValueMapProducer<pat::Jet>::setValue("Jet_leadTrackPt",ptMax);
-
-		//Fill vertex properties
-		VertexDistance3D vdist;
-		float maxFoundSignificance=0;
-		const auto & pv = (*pvs_)[0];
-	  	this->setValue("Jet_vtxPt",0);
-	   	this->setValue("Jet_vtxMass",0);
-    		this->setValue("Jet_vtx3dL",0);
-    		this->setValue("Jet_vtx3deL",0);
-    		this->setValue("Jet_vtxNtrk",0);
-
-		for(const auto &sv: *svs_){
-	      	GlobalVector flightDir(sv.vertex().x() - pv.x(), sv.vertex().y() - pv.y(),sv.vertex().z() - pv.z());
-	            GlobalVector jetDir(j.px(),j.py(),j.pz());
-                    if( Geom::deltaR2( flightDir, jetDir ) < 0.09 ){
-		        Measurement1D dl= vdist.distance(pv,VertexState(RecoVertex::convertPos(sv.position()),RecoVertex::convertError(sv.error())));
-			if(dl.significance() > maxFoundSignificance){
-				 maxFoundSignificance=dl.significance();
-				 this->setValue("Jet_vtxPt",sv.pt());
-				 this->setValue("Jet_vtxMass",sv.p4().M());
-				 this->setValue("Jet_vtx3dL",dl.value());
-				 this->setValue("Jet_vtx3deL",dl.error());
-				 this->setValue("Jet_vtxNtrk",sv.numberOfSourceCandidatePtrs());
-			}	
-		    } 
-		}
+        iEvent.getByToken(rhosrc_,rho_);
+
+      }
+
+    void fillAdditionalVariables(const pat::Jet&j)  override {
+
+    this->setValue("nPVs",pvs_->size());
+    this->setValue("rho",*(rho_.product()));
+
+    
+    float cone_boundaries[] = { 0.05, 0.1, 0.2, 0.3,0.4 }; 
+    size_t ncone_boundaries = sizeof(cone_boundaries)/sizeof(float);
+    std::vector<float> EmFractionEnergyRings(ncone_boundaries+1);
```

Comment:
start local variables and data members with lower case letter. Capitalization is reserved for class names.